timemachine/fe/plots.py

```diff
@@ -402,11 +411,14 @@ def plot_hrex_replica_state_distribution_heatmap(
     fig, ax = plt.subplots(figsize=figsize)
     p = ax.pcolormesh(replicas, states, fraction_by_replica_by_state)
 
-    if annotate:
+    # Skip text annotations if "auto" and large number of states
+    annotate_ = n_states <= 20 if annotate == "auto" else annotate
```

Comment:
nit: "auto" behavior does not depend on figsize, and there is 2x duplication of n_states threshold between HREX heatmap plots... not for this PR, but possible future enhancement: extract a function "plot_annotated_heatmap(matrix, figsize, ...)" or similar

Reply:
Changed the behavior in https://github.com/proteneer/timemachine/pull/1157/commits/bd48f8d2c13dc4fa32cb92012b14cb14cbf5f364 to allow overriding the default threshold, to avoid hardcoding implicit dependence on figsize.